### PR TITLE
Add a cheatsheet for Magit (Emacs Git major mode)

### PR DESCRIPTION
**Pull request by apg**
**What does your Instant Answer do?**

Displays a cheat sheet for [Magit](http://magit.vc/).

**What is the data source for your Instant Answer? (Provide a link if possible)**

http://daemianmack.com/magit-cheatsheet.html

**Why did you choose this data source?**

It's the cheat sheet that I always end up using.

**Are there any other alternative (better) data sources?**

The manual itself, but that sort of defeats the purpose of a cheat sheet.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Programmers / Emacs users.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

Not that I know of.

**Are you having any problems? Do you need our help with anything?**

Nope!

**Where did you hear about DuckDuckHack? (For first time contributors)**

I've been on the duckduckgo mailing list for years, and have just never contributed.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

magit cheat sheet

-----
IA Page: https://duck.co/ia/view/magit_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@apg Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

File: share/goodie/cheat_sheets/json/magit.json [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+      {
+        "key": "s",
+        "val": "Add untracked file to staging area"
+      },
+      {
+        "key": "i",
+        "val": "Add file to .gitignore"
+      },
+      {
+        "key": "C-u i",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/magit.json]: For any `key` that has successive key presses, like this, please wrap the separate keys in `[]` and join them with `,`:

```
"key": "[C-u], [i]"
```

That way they'll be visually separated when the IA is displayed. The existing Emacs Cheat Sheet demonstrates this: https://duckduckgo.com/?q=emacs+cheat+sheet&ia=cheatsheet

Comment (moollaza) [inline share/goodie/cheat_sheets/json/magit.json]: The JSON can be seen here https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/cheat_sheets/json/emacs.json

Comment (apg) [inline share/goodie/cheat_sheets/json/magit.json]: Ok. That's reasonable. Done.

On Tue, Oct 20, 2015 at 1:25 PM, Zaahir Moolla <notifications@github.com>
wrote:

> In share/goodie/cheat_sheets/json/magit.json
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1670#discussion_r42548713>
> :
>
> > +      {
> > +        "key": "M-1,2,3,4",
> > +        "val": "Expand all sections to the corresponding level of detail - 1, 2, 3 or 4"
> > +      }
> > +    ],
> > +    "Untracked Files": [
> > +      {
> > +        "key": "s",
> > +        "val": "Add untracked file to staging area"
> > +      },
> > +      {
> > +        "key": "i",
> > +        "val": "Add file to .gitignore"
> > +      },
> > +      {
> > +        "key": "C-u i",
>
> The JSON can be seen here
> https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/cheat_sheets/json/emacs.json
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1670/files#r42548713>
> .
>



-- 
http://www.apgwoz.com

**Comment by moollaza** [comment]
@apg this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@apg congrats! It's live: https://duckduckgo.com/?q=magit+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

Sorry for the late notice, it's actually been live for a little while now!

**Comment by apg** [comment]
I know! I used it yesterday! Thanks for merging!

On November 13, 2015 8:14:30 AM PST, Zaahir Moolla <notifications@github.com> wrote:
>@apg congrats! It's live:
>https://duckduckgo.com/?q=magit+cheat+sheet&ia=cheatsheet
>
>Thanks again for taking the time to contribute! We really appreciate
>it. Feel free to stick around and comment/help on other PR's, tackle
>open issues or even submit more Instant Answers!
>
>Sorry for the late notice, it's actually been live for a little while
>now!
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1670#issuecomment-156476412

-- 
Sent from space.

**Pull request merged**